opencog/attention/WARentCollectionAgent.cc

```diff
@@ -56,7 +57,7 @@ Handle WARentCollectionAgent::tournamentSelect(HandleSeq population){
         tournament[i] = population[idx];
     }
 
-    auto result = std::max_element(tournament, tournament + (_tournamentSize - 1), []
+    auto result = std::max_element(tournament, tournament + (sz - 1), []
```

Comment:
This can't happen as the population.size() is at least 1 otherwise the code exits before that. And this was causing the segfault during my testing.